poppler: 21.06.1 -> 22.01.0

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from https://repology.org/metapackage/poppler-glib/versions.


meta.description for poppler is: "A PDF rendering library"

meta.homepage for poppler is: "https://poppler.freedesktop.org/"

meta.changelog for poppler is: ""


###### Updates performed
- Version update

###### To inspect upstream changes

- [Poppler 21.7.0 released](https://lists.freedesktop.org/archives/poppler/2021-July/014803.html)
- [Poppler 21.8.0 released](https://lists.freedesktop.org/archives/poppler/2021-August/014836.html)
- [Poppler 21.9.0 released](https://lists.freedesktop.org/archives/poppler/2021-September/014870.html)
- [poppler-data 0.4.11 released](https://lists.freedesktop.org/archives/poppler/2021-September/014871.html)
- [Poppler 21.10.0 released](https://lists.freedesktop.org/archives/poppler/2021-October/014910.html)
- [Poppler 21.11.0 released](https://lists.freedesktop.org/archives/poppler/2021-November/014954.html)
- [Poppler 21.12.0 released](https://lists.freedesktop.org/archives/poppler/2021-December/014991.html)



###### Impact

<details>
<summary>
<b>Checks done</b> (click to expand)
</summary>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed

- 0 of 0 passed binary check by having a zero exit code.
- 0 of 0 passed binary check by having the new version present in output.
- found 21.12.0 with grep in /nix/store/m2rpqp544nhb4hppf4j0mdfvjcgkzy5g-poppler-glib-21.12.0
- found 21.12.0 in filename of file in /nix/store/m2rpqp544nhb4hppf4j0mdfvjcgkzy5g-poppler-glib-21.12.0

---

</details>
<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
747 total rebuild path(s)

595 package rebuild(s)

594 x86_64-linux rebuild(s)
153 x86_64-darwin rebuild(s)


First fifty rebuilds by attrpath
CuboCore.corepdf
adapta-gtk-theme
anystyle-cli
apacheHttpdPackages.mod_tile
apacheHttpdPackages_2_4.mod_tile
apvlv
arc-theme
arx-libertatis
asciidoc-full
asciidoc-full-with-plugins
auto-multiple-choice
ayu-theme-gtk
bookworm
btrbk
calibre
calligra
capitaine-cursors
chrome-gnome-shell
cinnamon.cinnamon-common
cinnamon.cinnamon-control-center
cinnamon.cinnamon-gsettings-overrides
cinnamon.cinnamon-screensaver
cinnamon.cinnamon-session
cinnamon.cinnamon-settings-daemon
cinnamon.iso-flags-png-320x420
cinnamon.iso-flags-svg
cinnamon.xreader
claws-mail
clevis
cloudcompare
colord
colord-gtk
cups-filters
darktable
dblatexFull
denemo
diff-pdf
diffoscope
diffpdf
digikam
disorderfs
docbookrx
dropbox-cli
eaglemode
emojione
engauge-digitizer
enlightenment.econnman
enlightenment.efl
enlightenment.enlightenment
enlightenment.ephoto
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from Cachix**:
```
nix-store -r /nix/store/m2rpqp544nhb4hppf4j0mdfvjcgkzy5g-poppler-glib-21.12.0 \
  --option binary-caches 'https://cache.nixos.org/ https://nix-community.cachix.org/' \
  --option trusted-public-keys '
  nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The Cachix cache is only trusted for this store-path realization.)
For the Cachix download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A poppler https://github.com/r-ryantm/nixpkgs/archive/94781b21dec834911d468d826e27a947cd3d40ef.tar.gz
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/m2rpqp544nhb4hppf4j0mdfvjcgkzy5g-poppler-glib-21.12.0
ls -la /nix/store/m2rpqp544nhb4hppf4j0mdfvjcgkzy5g-poppler-glib-21.12.0/bin
```

---

</details>
<br/>



### Pre-merge build results

NixPkgs review skipped

---

###### Maintainer pings

cc @ttuegel @jtojnar for [testing](https://github.com/ryantm/nixpkgs-update/blob/master/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

r-ryantm committed:
poppler: 21.06.1 → 22.01.0

- https://lists.freedesktop.org/archives/poppler/2021-July/014803.html
- https://lists.freedesktop.org/archives/poppler/2021-August/014836.html
- https://lists.freedesktop.org/archives/poppler/2021-September/014870.html
- https://lists.freedesktop.org/archives/poppler/2021-October/014910.html
- https://lists.freedesktop.org/archives/poppler/2021-November/014954.html
- https://lists.freedesktop.org/archives/poppler/2021-December/014991.html
- https://lists.freedesktop.org/archives/poppler/2022-January/015023.html

c++17 is now minimum

jtojnar committed:
poppler-data: clean up

https://lists.freedesktop.org/archives/poppler/2021-September/014871.html

jtojnar committed:
inkscape: fix build with Poppler 21.11.0

jtojnar commented:
@GrahamcOfBorg build inkscape cups-filters texlive scribusUnstable

risicle commented:
On macos 10.15 (cherry-picked to master):

```
builder for '/nix/store/h1z73iikvyhqp7r9z8fic3z5b5v6ls9x-gdal-3.3.2.drv' failed with exit code 2; last 10 log lines:
  /nix/store/3npg6a8nc5vpcyw98v085cmlz7f78kgs-bash-5.1-p12/bin/bash /private/tmp/nix-build-gdal-3.3.2.drv-0/source/gdal/libtool --mode=compile --silent --tag=CXX clang++ -I/private/tmp/nix-build-gdal-3.3.2.drv-0/source/gdal/port -I/private/tmp/nix-build-gdal-3.3.2.drv-0/source/gdal/gcore -I/private/tmp/nix-build-gdal-3.3.2.drv-0/source/gdal/alg -I/private/tmp/nix-build-gdal-3.3.2.drv-0/source/gdal/ogr -I/private/tmp/nix-build-gdal-3.3.2.drv-0/source/gdal/ogr/ogrsf_frmts -I/private/tmp/nix-build-gdal-3.3.2.drv-0/source/gdal/gnm -I/private/tmp/nix-build-gdal-3.3.2.drv-0/source/gdal/apps -DHAVE_AVX_AT_COMPILE_TIME -DHAVE_SSSE3_AT_COMPILE_TIME -DHAVE_SSE_AT_COMPILE_TIME -fpermissive  -Wall -Wextra -Winit-self -Wunused-parameter -Wformat -Werror=format-security -Wno-format-nonliteral -Wshorten-64-to-32 -Wshadow -Wmissing-include-dirs -Werror=vla -Wdate-time -Wnull-dereference -Wextra-semi -Wcomma -Wfloat-conversion -Wdocumentation -Wno-documentation-deprecated-sync -Wunused-private-field -Wmissing-declarations -Wnon-virtual-dtor -Woverloaded-virtual -fno-operator-names -Wzero-as-null-pointer-constant -Wimplicit-fallthrough  -I../vrt -I../mem -I../../ogr/ogrsf_frmts/mem -DGNM_ENABLED -I/private/tmp/nix-build-gdal-3.3.2.drv-0/source/gdal/port -I/nix/store/qlmzjna31i7w853l02jz5kfkc2734jn5-openjpeg-2.4.0-dev/include/openjpeg-2.4 -I/nix/store/yvz062ip2jsrzkkaqka6frrd7i3p5cjq-netcdf-4.8.0/include -I/nix/store/x3rbz8axlp3yjgkzypj79ddfaaxx842b-libpng-apng-1.6.37-dev -I/nix/store/x3rbz8axlp3yjgkzypj79ddfaaxx842b-libpng-apng-1.6.37-dev/include -I/nix/store/j4nm0ljg4fccf3j793nmzgpcif3hh1wm-libtiff-4.3.0-dev/include -I/nix/store/2in2vcnf5i7h6hb56d06xapfly56j4bb-libjpeg-turbo-2.1.0-dev -I/nix/store/2in2vcnf5i7h6hb56d06xapfly56j4bb-libjpeg-turbo-2.1.0-dev/include -I/nix/store/nb0j3bnywy8cfzam4yavlp6k26yd4410-zlib-1.2.11-dev -I/nix/store/nb0j3bnywy8cfzam4yavlp6k26yd4410-zlib-1.2.11-dev/include  -DGDAL_COMPILATION -DHAVE_POPPLER -DPOPPLER_MAJOR_VERSION=21 -DPOPPLER_MINOR_VERSION=12 -I/nix/store/f581n6jdssqydwyrz29j61i82facpvwl-poppler-glib-21.12.0-dev/include -I/nix/store/f581n6jdssqydwyrz29j61i82facpvwl-poppler-glib-21.12.0-dev/include/poppler   -DDO_NOT_USE_DEBUG_BOOL -c -o ../o/pdfdataset.lo pdfdataset.cpp
  pdfdataset.cpp:4985:16: error: no viable conversion from 'std::unique_ptr<GooString>' to 'GooString *'
      GooString* poMetadata = poCatalogPoppler->readMetadata();
                 ^            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  1 error generated.
  make[2]: *** [../../GDALmake.opt:648: ../o/pdfdataset.lo] Error 1
  make[2]: Leaving directory '/private/tmp/nix-build-gdal-3.3.2.drv-0/source/gdal/frmts/pdf'
  make[1]: *** [GNUmakefile:15: pdf-install-obj] Error 2
  make[1]: Leaving directory '/private/tmp/nix-build-gdal-3.3.2.drv-0/source/gdal/frmts'
  make: *** [GNUmakefile:114: frmts-target] Error 2
```

jtojnar commented:
@risicle Could you please try the patches mentioned in https://github.com/OSGeo/gdal/issues/4682?

risicle commented:
Success.

```
  patches = [
    (fetchpatch {
      name = "poppler-2.10-fix.part-1.patch";
      url = "https://github.com/OSGeo/gdal/commit/9c09870e374ca21d558101af3f4c09a6164fdfc3.patch";
      sha256 = "0dmf0jy7rniw7j4g5bwypwangbfxqvix03ydls4fw9nv94q6mrpz";
      stripLen = 1;
    })
    (fetchpatch {
      name = "poppler-2.10-fix.part-2.patch";
      url = "https://github.com/OSGeo/gdal/commit/53f969c9c287e4893ff8d9f97ddd1cb25c32dfb4.patch";
      sha256 = "1y43sqaxjx9sy18jryslifdmnwbhbhawn25gwx89nn4z46ighz4j";
      stripLen = 1;
    })
  ];
```

in case that saves anyone any time.

jtojnar commented:
Feel free to push it here. Otherwise I will try to do it on weekend (does it apply to `gdal`, `gdal_1_11` or `gdal_2`?)

risicle commented:
`gdal`

risicle commented:
No other new failures on macos 10.15.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/699

SuperSandro2000 commented on pkgs/applications/graphics/inkscape/default.nix:
```diff
@@ -82,6 +82,13 @@ stdenv.mkDerivation rec {
     })
+
+    # Fix build with Poppler 21.11.0.
+    # https://gitlab.com/inkscape/inkscape/-/commit/5724c21b9cb7b6176a7b36ca24068b148c817e82
```
```suggestion
```

jtojnar replied:
That was supposed to link to https://gitlab.com/inkscape/inkscape/-/merge_requests/3622

SuperSandro2000 commented on pkgs/development/libraries/poppler/default.nix:
```diff
@@ -34,13 +34,13 @@ let
 stdenv.mkDerivation rec {
   pname = "poppler-${suffix}";
-  version = "21.06.1"; # beware: updates often break cups-filters build, check texlive and scribusUnstable too!
+  version = "21.12.0"; # beware: updates often break cups-filters build, check texlive and scribusUnstable too!
```
Where does packages checked?

jtojnar replied:
What do you mean?

SuperSandro2000 replied:
Did we check if cups-filters, texlive and scribus still build?

jtojnar committed:
gdal: fix build with latest poppler

Poppler 22.01.0 requires std=c++17:

https://gitlab.freedesktop.org/poppler/poppler/-/merge_requests/1027

jtojnar committed:
scribusUnstable: fix build with latest poppler

Poppler 22.01.0 requires std=c++17:

https://gitlab.freedesktop.org/poppler/poppler/-/merge_requests/1027

It will be enforced in next release:

https://github.com/scribusproject/scribus/commit/0b1abeff0891b0907cdccb4ae965541b24b611f8
